arangod/Aql/IResearchViewOptimizerRules.cpp

```diff
@@ -291,6 +291,7 @@ bool attributesMatch(IResearchViewSort const& primarySort, IResearchViewStoredVa
                      latematerialized::NodeWithAttrsColumn& node,
                      std::unordered_map<ptrdiff_t, std::vector<ColumnVariant>>& usedColumnsCounter) {
```

Comment:
tbh, I don't really understand why do we need `std::unoredered_map<...>` here at all given all keys are generated sequentially, ofc there might be gaps, but `std::vector<...>` IMHO should be more efficient. 
Indeed that's more a question to @iurii-i-popov, but I suggest to consider switching to `std::vector<...>`

Reply:
I used to use a vector in another place. There is a problem with IResearchViewNode::SortColumnNumber. I put it to the end of the vector and got more complex code. That place was refactored and I was happy to remove the complexity there. You can also save sort column data in a separate variable.

You're welcome to change it.